### PR TITLE
Formatted filenames

### PR DESCRIPTION
**Pull request by phynes-sensiblecode**
Output files with requested name format.

### CODE REVIEW

**Review by SpencerHedger — CHANGES_REQUESTED** [review]
Just two issues that I could find.

File: bin/ons_csv_to_ctb_json_main.py [hunk 1]
```diff
@@ -58,26 +85,47 @@ def main():
     # A Cantabular dataset is equivalent to an ONS database.
     ctb_variables = build_ctb_variables(loader.classifications, loader.categories)
     ctb_datasets = build_ctb_datasets(loader.databases, ctb_variables)
-    filename = os.path.join(args.output_dir, 'dataset-metadata.json')
+
+    base_filename = output_filename(args.file_prefix, args.metadata_master_version,
```
[the remark this answers is not in the document]
Comment (phynes-sensiblecode) [reply bin/ons_csv_to_ctb_json_main.py:106]: I'll validate that all characters are either alphanumeric or one of `"-"`, `"_"`, `"."`, `" "`

File: bin/ons_csv_to_ctb_json_main.py [hunk 2]
```diff
@@ -58,26 +85,47 @@ def main():
...
+                                    FILE_CONTENT_TYPE_DATASET, todays_date,
+                                    args.build_number)
```
[the remark this answers is not in the document]
Comment (phynes-sensiblecode) [reply bin/ons_csv_to_ctb_json_main.py:108]: I'll check for numbers greater than or equal to 0

**Review by SpencerHedger — APPROVED** [review]
Looks good.

**Pull request merged**